src/main/java/org/lantern/multicast/LanternMulticast.java

```diff
@@ -0,0 +1,141 @@
+package org.lantern.multicast;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.lantern.JsonUtils;
+import org.lantern.LanternUtils;
+import org.lantern.event.Events;
+import org.lantern.state.StaticSettings;
+import org.lantern.state.SyncPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+
+/**
+ * Uses multicast to detect other Lanterns on the local network.
+ */
+public class LanternMulticast {
+    
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    public static final int MC_PORT = 9864;
+    
+    public static final String MC_ADDR = "232.77.77.77";
+    
+    private final int sendPort;
+    
+    
+    private final Collection<String> endpoints = new HashSet<String>();
+    
+    public LanternMulticast() {
+        this(MC_PORT);
+    }
+
+    public LanternMulticast(final int port) {
+        this.sendPort = port;
+    }
+    
+
+    public void join() {
+        try {
+            final InetAddress group = InetAddress.getByName(MC_ADDR);
+            final MulticastSocket ms = new MulticastSocket(MC_PORT);
+            ms.joinGroup(group);
+            
+            if (LanternUtils.isLanternPi()) {
+                sendHellos(group, ms);
+                addShutdownHook(group, ms);
+            }
+            
+            listen(ms);
```

Comment:
Do we even need to listen on the Pi side? Whenever the "hello" and "bye" messages get sent, we end up receiving these locally as well.